regression with `if parent_type(LL[0]) == syms.test:`

looks like this regressed as part of #1041

CC @asfaltboy

here's some code, extracted from black:

```python
if parent_type(LL[0]) == syms.test:
    pass
```

before it raised no errors, now it raises:

```console
$ python3 /tmp/pycodestyle/pycodestyle.py t.py 
t.py:1:11: E721 do not compare types, use 'isinstance()'
```

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -136,7 +136,7 @@ COMPARE_NEGATIVE_REGEX = re.compile(r'\b(?<!is\s)(not)\s+[^][)(}{ ]+\s+'
                                     r'(in|is)\s')
 COMPARE_TYPE_REGEX = re.compile(
     r'(?:[=!]=|is(?:\s+not)?)\s+type(?:\s*\(\s*([^)]*[^ )])\s*\))' +
-    r'|type(?:\s*\(\s*([^)]*[^ )])\s*\))\s+(?:[=!]=|is(?:\s+not)?)'
+    r'|\btype(?:\s*\(\s*([^)]*[^ )])\s*\))\s+(?:[=!]=|is(?:\s+not)?)'
 )
 KEYWORD_REGEX = re.compile(r'(\s*)\b(?:%s)\b(\s*)' % r'|'.join(KEYWORDS))
 OPERATOR_REGEX = re.compile(r'(?:[^,\s])(\s*)(?:[-+*/|!<=>%&^]+|:=)(\s*)')

--- a/testsuite/E72.py
+++ b/testsuite/E72.py
@@ -83,3 +83,8 @@ from . import custom_types as types
 
 red = types.ColorTypeRED
 red is types.ColorType.RED
+#: Okay
+from . import compute_type
+
+if compute_type(foo) == 5:
+    pass

